src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/TestUriHelper.cs

```diff
@@ -23,27 +23,22 @@ public static Uri BuildTestUri(string hint)
             else
             {
                 var uriHint = new Uri(hint);
-                return new UriBuilder(uriHint) { Port = FindFreePort(uriHint.Port) }.Uri;
+                if(uriHint.Port == 0)
+                {
+                    return new UriBuilder(uriHint) { Port = FindFreePort() }.Uri;
+                }
+                else
+                {
+                    return uriHint;
+                }
             }
         }
 
         public static int FindFreePort()
```

Comment:
I would copy the method from Kestrel for consistency, we're already using this method in two places.  The code is exactly the same as what you have here.

https://github.com/aspnet/KestrelHttpServer/blob/dev/test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs#L508